cmd/src/update.go

```diff
@@ -0,0 +1,156 @@
+
+		$ src update
+`
+
+	flagSet := flag.NewFlagSet("update", flag.ExitOnError)
+
+	var apiFlags = api.NewFlags(flagSet)
+
+	handler := func(args []string) error {
+		currentDirectory, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		err = isDirWritable(currentDirectory)
+		if err != nil {
+			runWithElevatedPrivilege()
+			return nil
+		}
+
+		fmt.Printf("Current version: %s\n", buildTag)
+
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+		recommendedVersion, err := getRecommendedVersion(context.Background(), client)
+		if err != nil {
+			return err
+		}
+		if recommendedVersion == "" {
+			fmt.Println("Recommended Version: <unknown>")
+			return nil
+		}
+		if buildTag == recommendedVersion {
+			fmt.Printf("src-cli is already at the recommended version: %s\n", recommendedVersion)
+			return nil
+		}
+
+		// Retrieving latest release information.
+		updateManager := &update.Manager{
+			Command: "src",
+			Store: &github.Store{
+				Owner:   "sourcegraph",
+				Repo:    "src-cli",
+				Version: buildTag,
+			},
+		}
+		if runtime.GOOS == "windows" {
+			updateManager.Command += ".exe"
+		}
+		releases, err := updateManager.LatestReleases()
+		if err != nil {
+			return err
+		}
+		if len(releases) == 0 {
+			return fmt.Errorf("No latest src-cli release update")
+		}
+		latestRelease := releases[0]
+		if latestRelease.Version != recommendedVersion {
+			return fmt.Errorf("Mismatch of recommended version")
+		}
+
+		updateAsset := latestRelease.FindTarball(runtime.GOOS, runtime.GOARCH)
+		if updateAsset == nil {
+			return fmt.Errorf("src-cli binary has not been installed in this machine")
+		}
+		fmt.Printf("Downloading src-cli version %s\n", recommendedVersion)
+		tarball, err := updateAsset.DownloadProxy(progress.Reader)
+		if err != nil {
+			return fmt.Errorf("Failed downloading src-cli release: %v\n", err)
+		}
+
+		bin, err := exec.LookPath(updateManager.Command)
+		if err != nil {
+			return fmt.Errorf("Failed looking up path of %v\n", updateManager.Command)
+		}
+		dir := filepath.Dir(bin)
+
+		if runtime.GOOS == "windows" {
+			// Windows need to use absolute path for the file.
+			if dir == "." {
+				currDir, err := os.Getwd()
+				if err != nil {
+					return fmt.Errorf("Failed detecting current directory: %v\n", err)
+				}
+				dir = currDir
+			}
+
+			// Workaround for Windows to rename the old file first.
+			dst := filepath.Join(dir, updateManager.Command)
+			old := dst + ".old"
+			if err := os.Rename(dst, old); err != nil {
+				return fmt.Errorf("Windows renaming\n")
+			}
+		}
+
+		if err := updateManager.InstallTo(tarball, dir); err != nil {
+			return fmt.Errorf("Failed installing src-cli: %v\n", err)
+		}
+
+		fmt.Printf("src-cli has been updated from version %s to %s\n", buildTag, recommendedVersion)
+
+		// Keep the terminal installation window open after Windows' UAC privilege elevation.
+		if runtime.GOOS == "windows" {
+			fmt.Println("Press Enter key to exit")
```

Comment:
Isn't the famous any key enough here? 😄 
![image](https://user-images.githubusercontent.com/1185253/106123111-cdf06700-6159-11eb-8c9e-97605ca13e91.png)

```suggestion
			fmt.Println("Press any key to exit")
```